scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/TasksPaginationController.java

```diff
@@ -57,4 +58,104 @@ public Layout buildView() {
         return this.view.build();
     }
 
+    /**
+     * Fetch the next item list page
+     */
+    public void nextPage() {
+        model.setPage(model.getPage() + 1);
+        this.fetch(false);
+    }
+
+    @Override
+    public void previousPage() {
+        int curPage = model.getPage();
+        if (curPage == 0)
+            return;
+        model.setPage(curPage - 1);
+        this.fetch(false);
+    }
+
+    @Override
+    public void firstPage() {
+        model.setPage(0);
+        this.fetch(false);
+    }
+
+    public void lastPage() {
+        this.model.setPage(this.model.getMaxPage());
+        this.fetch(false);
+    }
+
+    public void goToPage(int pageNumber) {
+        if (pageNumber < 0) {
+            pageNumber = 0;
+        }
+
+        int maxPage = this.model.getMaxPage();
+        if (pageNumber > maxPage) {
+            pageNumber = maxPage;
+        }
+
+        this.model.setPage(pageNumber);
+        this.fetch(false);
+    }
+
+    public void computeMaxPage(long nbItem) {
+        this.model.setTotalItems(nbItem);
+    }
+
+    /**
+     * Gets the text that displays the pagination status.
+     * @return the text that displays the pagination status.
+     */
+    public String getPaginationRangeLabel() {
+        int page = this.model.getPage();
+        int size = this.model.getPageSize();
+
+        long index = page * size;
```

Comment:
Can you compose this method into smaller methods where each method name explains what it does